src/mono/wasi/Wasi.Build.Tests/BuildThenRun.cs

```diff
@@ -0,0 +1,109 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+using Wasm.Build.Tests;
+
+#nullable enable
+
+namespace Wasi.Build.Tests;
+
+public class BuildThenRunTests : BuildTestBase
```

Comment:
`BuildThenRun`, and `PublishThenRun` will be the most common scenarios anyway, and that does not help differentiate the tests in a meaningful way.
`BuildThenPublish` is different because it is specifically about build behavior when doing a `build` followed by a `publish` because the latter has overlap with the former.

The basic tests can still be build/publish for single-file, and non-single-file bundle cases. The TZ one can be in a new file.